src/components/Filters/FiltersDialog.tsx

```diff
@@ -0,0 +1,312 @@
+									onClick={onAddClick}
+									startIcon={<FontAwesomeIcon icon={faPlus} />}
+								>
+									{t('actions.add_filter_in_or')}
+								</Button>
+							)}
+							{index !== 0 && (
+								<Button
+									aria-label={t('actions.remove_filter')}
+									type="button"
+									onClick={element.onDropIndexClick(element.index)}
+									color="error"
+									startIcon={<FontAwesomeIcon icon={faTrash} />}
+								>
+									Remove
+								</Button>
+							)}
+						</Box>
+					</Box>
+				))}
+		</Box>
+	);
+};
+
+const widgets = {
+	CheckboxWidget: SelectWidget,
+};
+
+const initialFormData = [
+	{
+		field: undefined,
+		operator: undefined,
+		value: undefined,
+	},
+];
+
+const normalizeFormData = (
+	data: FormData[] | FormData | undefined,
+	schema: JSONSchema,
+) => {
+	if (!data || !Array.isArray(data)) {
+		return data;
+	}
+	return data.map((d) => {
+		if (!schema.properties) {
+			return d;
+		}
+		const field = d?.field ?? Object.keys(schema.properties)[0];
+		const propertySchema = getPropertySchema(field, schema);
+		const prefix =
+			propertySchema?.type === 'array' ? 'items.properties.' : 'properties.';
+		const refSchema = propertySchema
+			? getRefSchema(propertySchema, prefix)
+			: propertySchema;
+		const model = getDataModel(refSchema);
+		const operator = model
+			? Object.keys(model?.operators).find((o) => o === d?.operator) ??
+			  Object.keys(model?.operators)[0]
+			: undefined;
+		return {
+			field: d?.field ?? field,
+			operator,
+			value: d?.value,
+		};
+	});
+};
+
+interface FiltersDialogProps {
+	open: boolean;
+	schema: JSONSchema;
+	onClose: ((filter?: JSONSchema | null) => void) | undefined;
+	data?: FormData[] | FormData;
+}
+
+export const FiltersDialog = ({
+	schema,
+	open,
+	data = initialFormData,
+	onClose,
+}: FiltersDialogProps) => {
+	const { t } = useTranslation();
+	const [formData, setFormData] = React.useState<
+		FormData[] | FormData | undefined
+	>(normalizeFormData(data, schema));
+
+	const handleChange = React.useCallback(
+		({ formData: data }: IChangeEvent<FormData[]>) => {
+			// Unfortunately we cannot detect which filed is changing so we need to set a previous state
+			// github.com/rjsf-team/react-jsonschema-form/issues/718
+			const newFormData = Array.isArray(formData)
+				? data.map((d, i) => {
+						if (
+							formData &&
+							formData[i]?.field &&
+							formData[i]?.operator &&
+							(d.field !== formData[i].field ||
+								d.operator !== formData[i].operator)
+						) {
+							return { ...d, value: undefined };
+						}
+						return d;
+				  })
+				: formData;
+
+			setFormData(normalizeFormData(newFormData, schema));
+		},
+		[setFormData, formData],
+	);
+
+	const handleSubmit = ({ formData }: IChangeEvent<FormData[] | FormData>) => {
+		const filters = Array.isArray(formData)
+			? createFilter(schema, formData)
+			: formData.value
+			? createFullTextSearchFilter(schema, formData.value)
+			: null;
+		onClose?.(filters);
```

Comment:
To avoid the extra calls if `onClose` is nullish?
```suggestion
		onClose?.(Array.isArray(formData)
			? createFilter(schema, formData)
			: formData.value
			? createFullTextSearchFilter(schema, formData.value)
			: null);
```

Reply:
I would probably add an if !onClose return in this case, as I feel it is too nested otherwise